peachjam_search/views.py

```diff
@@ -753,3 +770,99 @@ def get(self, request, *args, **kwargs):
+                filters=urlencode(params, doseq=True)
+            ).get_sorted_filters_string()
+            saved_search = SavedSearch.objects.filter(
+                user=self.request.user, q=q, filters=filters
+            ).first()
+            if saved_search:
+                return HttpResponseRedirect(
+                    reverse(
+                        "search:saved_search_update", kwargs={"pk": saved_search.pk}
+                    )
+                )
+            return HttpResponseRedirect(
+                reverse("search:saved_search_create") + f"?q={q}&{filters}"
+            )
+        context = super().get_context_data(**kwargs)
+        return self.render_to_response(context)
+
+
+class SavedSearchCreateView(CreateView):
+    permission_required = "peachjam_search.add_savedsearch"
+    template_name = "peachjam_search/saved_search_form.html"
+    model = SavedSearch
+    form_class = SavedSearchForm
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        if self.request.user.is_authenticated:
+            instance = SavedSearch()
+            instance.user = self.request.user
+            instance.last_alert = now()
+            instance.q = self.request.GET.get("q", "")
+            instance.filters = self.request.GET.urlencode()
+            instance.filters = instance.get_sorted_filters_string()
+            kwargs["instance"] = instance
+        return kwargs
+
+    def get_success_url(self):
+        return reverse("search:saved_search_update", kwargs={"pk": self.object.pk})
+
+
+class SavedSearchUpdateView(UpdateView):
+    permission_required = "peachjam_search.change_savedsearch"
+    template_name = "peachjam_search/saved_search_form.html"
+    model = SavedSearch
+    form_class = SavedSearchForm
+    context_object_name = "saved_search"
+
+    def get_queryset(self):
+        return self.request.user.saved_searches.all()
+
+    def get_success_url(self):
+        return reverse("search:saved_search_update", kwargs={"pk": self.object.pk})
+
+
+class SavedSearchListView(ListView):
+    permission_required = "peachjam_search.view_savedsearch"
+    template_name = "peachjam_search/saved_search_list.html"
+    model = SavedSearch
+    context_object_name = "saved_searches"
+
+    def get_queryset(self):
+        return self.request.user.saved_searches.all()
+
+
+class SavedSearchDeleteView(DeleteView):
+    permission_required = "peachjam_search.delete_savedsearch"
+    model = SavedSearch
+    success_url = reverse_lazy("search:saved_search_list")
+
+    def get_queryset(self):
+        return self.request.user.saved_searches.all()
+
+    def get_template_names(self):
+        if self.request.htmx:
+            return ["peachjam_search/_saved_search_list.html"]
+        return ["peachjam_search/saved_search_list.html"]
+
+    def post(self, *args, **kwargs):
+        self.object = self.get_object()
+        self.object.delete()
+        response = HttpResponse("saved search deleted")
```

Comment:
I think it's simpler not to use htmx at all here, there's no benefit, reloading the whole page is quick and deleting isn't a common operation. Less fuss.

Reply:
we can maintain htmx if we plan on having a delete button on the search page modal,